src/utils/deployFullvm.ts

```diff
@@ -0,0 +1,68 @@
+import type { default as Fullvm } from "../types/fullvm";
+import type { Disk, Env } from "../types/vm"
+import createNetwork from "./createNetwork";
+const { DiskModel, MachineModel, MachinesModel } =
+  window.configs?.grid3_client ?? {};
+import type { IProfile } from "../types/Profile";
+import deploy from "./deploy";
+import checkVMExist from "./prepareDeployment";
+import { Network } from "../types/kubernetes";
+
+export default async function deployFullvm(data: Fullvm, profile: IProfile) {
+  const { disks, envs, rootFs, ...base } = data;
+  const { name, flist, cpu, memory, entrypoint, network: nw } = base;
+  const { publicIp, publicIp6, planetary, nodeId } = base;
+
+  // VM Specs
+  const vm = new MachineModel();
+  vm.name = name; //`vm${randomSuffix}`;
+  vm.node_id = nodeId;
+  vm.disks = disks.map(createDisk);
+  vm.public_ip = publicIp;
+  vm.public_ip6 = publicIp6;
+  vm.planetary = planetary;
+  vm.cpu = cpu;
+  vm.memory = memory;
+  vm.rootfs_size = rootFs;
+  vm.flist = flist;
+  vm.entrypoint = entrypoint;
+  vm.env = createEnvs(envs);
+
+  // VMS Specs
+  const vms = new MachinesModel();
+  vms.name = name;
+  vms.network = createNetwork(new Network());
+  vms.machines = [vm];
+
+  const metadate = {
+    "type": "vm",
+    "name": name,
+    "projectName": "Fullvm"
+  };
+  vms.metadata = JSON.stringify(metadate);
+
+
+  return deploy(profile, "Fullvm", name, async (grid) => {
+    await checkVMExist(grid, "fullvm", name);
+    return grid.machines
+      .deploy(vms)
+      .then(() => grid.machines.getObj(name))
+      .then(([vm]) => vm);
+  });
+}
+
+function createDisk({ name, size, mountpoint }: Disk) {
+  const disk = new DiskModel();
+  disk.name = name;
+  disk.size = size;
+  disk.mountpoint = mountpoint;
+  return disk;
+}
+
+function createEnvs(envs: Env[]): { [key: string]: string } {
+  return envs.reduce((res, env) => {
+    res[env.key] = env.value;
+    return res;
+  }, {});
+}
+
```

Comment:
Why we didn't use deployVM instead of creating a new file?

Reply:
There were some requested changes as adding a default disk for the full VM and I preferred to do it in a separate file. Do you think it's better if I made it in the same file?

Reply:
done.